genesis/process/intermediate/delegationDeployProcessor.go

```diff
@@ -0,0 +1,52 @@
+package intermediate
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ElrondNetwork/elrond-go/genesis"
+)
+
+const stakedPlaceholder = "%sc_total_stake%"
+
+var zero = big.NewInt(0)
+
+type delegationDeployProcessor struct {
+	*deployProcessor
+	accountsParser genesis.AccountsParser
+	nodePrice      *big.Int
+}
+
+func NewDelegationDeployProcessor(
```

Comment:
why delegation and deploy ? the contract was not deployed before - and here you need only to send the total data to the staking smart contract.

Reply:
I had to create 2 deploy processors: one used for simple contracts (like answer.wasm) and one used for the delegation SC. The reason behind this is that the delegation SC **has to know** on init about the total amount of stake (I'm still against this, could have been resolved in some other manner) and that is why it has been implemented like this.

Reply:
I think this could have been handled the general "migration style" where you have a single script for deployment and initial state build. Most contracts will have constructor parameters (unless for the ones we deploy at the beginning we make an internal decisions that we will follow a no constructor parameters patterns). So if the list of initial contracts will increase, probably we will have to build more of the deploy/processor pairs

Reply:
Agree, I also dislike this pattern, we need a processor for each type of SC that has some "extra"  parameters on the constructor. Not my decision, I have just implemented a way so that contract can be called at genesis time.

Reply:
Will try to improve this in the next iteration (of the delegation SC)

Reply:
Init is automatically called at deploy. Delegation should have number of nodes as init, not stake. But that is another topic